### PR TITLE
feat: telegram login

### PR DESCRIPTION
**Pull request by gregfromstl**
## TL;DR

This PR introduces support for Telegram OAuth across various components of the thirdweb packages.

## What changed?

- Added 'Telegram' as an OAuth option in `WALLETS` array within the `constants.ts` file.
- Introduced `telegramIconUri` constant and its usage across the application.
- Updated `socialIcons` object and methods to handle new Telegram icon.
- Enhanced `WalletImage` components to recognize and display the Telegram icon.
- Included Telegram in OAuth login path determination within `oauthSignIn.ts`.
- Updated enumerations and type definitions to include Telegram.
- Modified `InAppNativeConnector` and `InAppWebConnector` to handle Telegram OAuth redirection.

## How to test?

1. Navigate to the authentication section in the application.
2. Check for the new 'Sign in with Telegram' option.
3. Attempt to sign in using a Telegram account and verify the successful login flow.

## Why make this change?

To expand the authentication options available to users, providing more flexibility and convenience. Telegram is a popular messaging app, and integrating it as an OAuth option can enhance user experience.

---

---

## PR-Codex overview
The focus of this PR is to add support for Telegram authentication across different modules.

### Detailed summary
- Added `telegram` as a social authentication option
- Included `TELEGRAM` in `OauthOption` type
- Updated various modules to support Telegram authentication
- Added Telegram icon URIs for different platforms

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by gregfromstl** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3830?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3830** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3830?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3827** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3827?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gregfromstl and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/playground-web/src/lib/constants.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export const WALLETS = [
       options: [
         "google",
         "discord",
-        "apple",
+        "telegram",
```
Comment (jnsdls) [inline apps/playground-web/src/lib/constants.ts:13]: no more apple?

**Review by jnsdls — APPROVED** [review]
add a changeset then LGTM

File: apps/playground-web/src/lib/constants.ts (hunk 1 above)
Comment (jnsdls) [reply apps/playground-web/src/lib/constants.ts:13]: playground, gotcha

**Pull request merged**